components/remote_settings/src/storage.rs

```diff
@@ -60,39 +101,373 @@ impl Storage {
     /// passed to `set_attachment`.
     pub fn get_attachment(
         &self,
-        _collection_url: &str,
-        _attachment_id: &str,
+        collection_url: &str,
+        attachment_id: &str,
     ) -> Result<Option<Attachment>> {
-        Ok(None)
+        let conn = self.conn.lock().unwrap();
+        let mut stmt =
+            conn.prepare("SELECT data FROM attachments WHERE id = ? AND collection_url = ?")?;
+        let result: Option<Vec<u8>> = stmt
+            .query_row(params![attachment_id, collection_url], |row| row.get(0))
+            .optional()?;
+        if let Some(data) = result {
+            let attachment: Attachment = serde_json::from_slice(&data)?;
+            Ok(Some(attachment))
+        } else {
+            Ok(None)
+        }
     }
 
     /// Set the list of records stored in the database, clearing out any previously stored records
     pub fn set_records(
         &self,
-        _collection_url: &str,
+        collection_url: &str,
         records: &[RemoteSettingsRecord],
     ) -> Result<()> {
-        for record in records {
-            println!("Should store record: {record:?}");
+        let mut conn = self.conn.lock().unwrap();
+
+        let tx = conn.transaction()?;
+
+        // Delete existing records for this collection_url
```

Comment:
Maybe this should delete records, regardless of collection_url.   If that changes, then I don't think there's any use in keeping around the old records.  I'm not totally sure on this one though, what's your opinion?

Reply:
I am not 100% sure about when this is used. Based on the signature of the function, I would assume we insert records just for this one particular `collection_url`, and therefore we decide to clear out old entries. If we clear out all other `collection_urls` as well, then... what happens if we call this method 5 times for example with different `collection_urls`? Or can one instance just have one `collection_url`?

Reply:
..and because it's kind of a hack, I'm hoping to prevent corner cases as best as we can.  It feels to me like there would be less if we deleted the old data, but I could definitely be wrong.

Reply:
Ok! What happens to the metadata though? Should we delete all previous metadata as well? Scenario (maybe unlikely but have to account for that anyway)

* Store records for `url1`
* Afterwards, store records for `url2`

State of the `db`: Just regords for `url2` because we delete all other entries.

What happens to a) the `attachements`? I assume they stay. And I introduced a `collection_metadata` where we store the `last_modified` and `fetched true/false`. Should we delete all entries for `url1` there as well?

Reply:
I updated the PR to reflect that. We delete all `records` and `collection_metadata` entries when setting new `records`.